main.go

```diff
@@ -127,20 +87,26 @@ func mainCmd(args []string) int {
 
 func parseFlags(checker *errcheck.Checker, args []string) ([]string, int) {
 	flags := flag.NewFlagSet(args[0], flag.ContinueOnError)
-	flags.BoolVar(&checker.Blank, "blank", false, "if true, check for errors assigned to blank identifier")
-	flags.BoolVar(&checker.Asserts, "asserts", false, "if true, check for ignored type assertion results")
-	flags.BoolVar(&checker.WithoutTests, "ignoretests", false, "if true, checking of _test.go files is disabled")
-	flags.BoolVar(&checker.WithoutGeneratedCode, "ignoregenerated", false, "if true, checking of files with generated code is disabled")
+
+	var checkAsserts, checkBlanks bool
+
+	flags.BoolVar(&checkBlanks, "blank", false, "if true, check for errors assigned to blank identifier")
+	flags.BoolVar(&checkAsserts, "asserts", false, "if true, check for ignored type assertion results")
+	flags.BoolVar(&checker.Exclusions.TestFiles, "ignoretests", false, "if true, checking of _test.go files is disabled")
+	flags.BoolVar(&checker.Exclusions.GeneratedFiles, "ignoregenerated", false, "if true, checking of files with generated code is disabled")
 	flags.BoolVar(&checker.Verbose, "verbose", false, "produce more verbose logging")
 
 	flags.BoolVar(&abspath, "abspath", false, "print absolute paths to files")
 
 	tags := tagsFlag{}
 	flags.Var(&tags, "tags", "comma or space-separated list of build tags to include")
 	ignorePkg := flags.String("ignorepkg", "", "comma-separated list of package paths to ignore")
-	ignore := ignoreFlag(map[string]*regexp.Regexp{})
-	flags.Var(ignore, "ignore", "[deprecated] comma-separated list of pairs of the form pkg:regex\n"+
+	ignore := flags.String("ignore", "", "[deprecated] comma-separated list of pairs of the form pkg:regex\n"+
 		"            the regex is used to ignore names within pkg.")
+	if *ignore != "" {
+		fmt.Fprintf(os.Stderr, "error: -ignore flag is deprecated, please use -exclude and/or -ignorepkg instead")
+		return nil, exitFatalError
+	}
```

Comment:
Can you alias -ignore to another flag, or trivially transform it somehow?

Reply:
I don't see how I can do it without adding a special field to Exclusions struct. I tried making a special case in logic that processes Packages, however for that, -ignore would had to treat the absence of `:<regexp>`portion at the end of each of the strings in the list as just the package name, whereas it does entirely the opposite: it assumes entire string is the regexp per https://github.com/kisielk/errcheck/blob/master/main.go#L46

Reply:
My concern is that it would break peoples' workflows to make this change. I know that it's been deprecated, but I feel that we should keep it, or go to 2.0.

Reply:
Functionality restored, PTAL